Trustroots/trustroots
# Convert codebase to ES6

#### Proposed Changes

* Convert all code (server and client) to ES6:
  * Manually fix a few var → let/const situations
  * Remove [`use strict`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode) using https://github.com/5to6/5to6-codemod and enable `no-strict` rule
  * Convert `var` → `const`/`let` (using `eslint --fix` and a few manual fixes)
  * Other things that `eslint --fix` can do
* Reorganise testutils to server and client subfolders for easier linting via separate server & client side overrides
* Update Eslint config:
  *  Lint everything as es6
  * Refactor to be more specific for each override section; each section has their own globals instead of keeping them together for the whole app
  * Separate client-side and server-side, as well tests and app files.
  *  Use `eslint:recommended` as a base; remove rules in our custom set which would be duplicate with recommended.
  * Add `one-var` linter rule since we're already touching all these lines. This should make e.g. automatically replacing require→import easier later on.
   * Removed all redundant Eslint disabled rules


#### Testing Instructions

* Please review `.eslintrc.js` — there are so many other changes that you might just want to skim through a bit and smoke test the site. 
* `npm start` works?
* `npm run lint` green?
* Confirm that clientside built bundles don't have `const`/`let` in sources and that they're ES5 code.
